config/confighttp/confighttp.go

```diff
@@ -340,7 +341,11 @@ func (hss *ServerConfig) ToServer(_ context.Context, host component.Host, settin
 		o(serverOpts)
 	}
 
-	handler = httpContentDecompressor(handler, serverOpts.errHandler, serverOpts.decoders)
+	if hss.MaxRequestBodySize == 0 {
```

Comment:
Should this check if (for whatever reason it happens) if the size is less than or equal to zero?